Sources/BraveWallet/Crypto/Stores/CryptoStore.swift

```diff
@@ -285,24 +291,23 @@ public class CryptoStore: ObservableObject {
       let pendingTransactions = await fetchPendingTransactions()
       var newPendingRequest: PendingRequest?
       if !pendingTransactions.isEmpty {
-        if self.buySendSwapDestination != nil {
-          // Dismiss any buy send swap open to show the unapproved transactions
-          self.buySendSwapDestination = nil
-        }
         newPendingRequest = .transactions(pendingTransactions)
       } else { // no pending transactions, check for webpage requests
         newPendingRequest = await fetchPendingWebpageRequest()
       }
       // Verify this new `newPendingRequest` isn't the same as the current
       // `pendingRequest` because re-assigning the same request could cause
       // present of a previously dismissed / ignored pending request (#6750).
-      if newPendingRequest?.id != self.pendingRequest?.id {
-        self.pendingRequest = newPendingRequest
+      guard newPendingRequest?.id != self.pendingRequest?.id else { return }
+      if self.buySendSwapDestination != nil && newPendingRequest != nil {
+        // Dismiss any buy send swap open to show the newPendingRequest
+        self.buySendSwapDestination = nil
       }
       // If we set these before the send or swap screens disappear for some reason it may crash
       // within the SwiftUI runtime or fail to dismiss. Delaying it to give time for the
       // animation to complete fixes it.
       DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
+        self.pendingRequest = newPendingRequest
         self.isPresentingPendingRequest = self.pendingRequest != nil
```

Comment:
i think you can just move this assignment inside `pendingRequest`'s didSet

Reply:
Updated in [18691db](https://github.com/brave/brave-ios/pull/6878/commits/18691db606adc71c978f72e9b2e287f7cf9773cb)